cloud/scope/powervs_machine_test.go

```diff
@@ -190,6 +191,254 @@ func TestNewPowerVSMachineScope(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockpowervs = mock.NewMockPowerVS(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	t.Run("SetRegion", func(t *testing.T) {
+		t.Run("Set region", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			scope := setupPowerVSMachineScope(clusterName, machineName, core.StringPtr(pvsImage), core.StringPtr(pvsNetwork), true, mockpowervs)
+			region := "us-south"
+			scope.SetRegion(region)
+			require.Equal(t, scope.GetRegion(), region)
+		})
+		t.Run("No region set", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			scope := setupPowerVSMachineScope(clusterName, machineName, core.StringPtr(pvsImage), core.StringPtr(pvsNetwork), true, mockpowervs)
+			require.Equal(t, scope.GetRegion(), "")
+		})
+	})
+}
+
+func TestSetZone(t *testing.T) {
+	var (
+		mockpowervs *mock.MockPowerVS
+		mockCtrl    *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockpowervs = mock.NewMockPowerVS(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	t.Run("SetZone", func(t *testing.T) {
+		t.Run("Set zone", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			scope := setupPowerVSMachineScope(clusterName, machineName, core.StringPtr(pvsImage), core.StringPtr(pvsNetwork), true, mockpowervs)
+			region := "us-south-1"
+			scope.SetZone(region)
+			require.Equal(t, scope.GetZone(), region)
+		})
+		t.Run("No zone set", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			scope := setupPowerVSMachineScope(clusterName, machineName, core.StringPtr(pvsImage), core.StringPtr(pvsNetwork), true, mockpowervs)
+			require.Equal(t, scope.GetZone(), "")
+		})
+	})
+}
+
+func TestSetInstanceState(t *testing.T) {
+	var (
+		mockpowervs *mock.MockPowerVS
+		mockCtrl    *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockpowervs = mock.NewMockPowerVS(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	t.Run("SetInstanceState", func(t *testing.T) {
+		t.Run("Set Instance state to ready", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			scope := setupPowerVSMachineScope(clusterName, machineName, core.StringPtr(pvsImage), core.StringPtr(pvsNetwork), true, mockpowervs)
+			instanceState := core.StringPtr("Ready")
+			scope.SetInstanceState(instanceState)
+			require.Equal(t, scope.GetInstanceState(), infrav1beta2.PowerVSInstanceState(*instanceState))
+		})
+	})
+}
+
+func TestSetProviderID(t *testing.T) {
+	var (
+		mockpowervs *mock.MockPowerVS
+		mockCtrl    *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockpowervs = mock.NewMockPowerVS(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	t.Run("Set Provider ID", func(t *testing.T) {
+		providerID := core.StringPtr("foo-provider-id")
+		t.Run("Set Provider ID in v2 format", func(t *testing.T) {
```

Comment:
May be we can add with nil id as well. Just to cover all possible cases.

Reply:
Done